syntax = "proto3";

package Seraphim.Face.FaceRecognizer;

import "Types.proto";

/*
 * Requests
 *
 * Client --> Server
 */

message TrainingRequest {
  Types.Image2D image = 1;
  int32 label = 2;
  bool invalidate = 3;
}

message PredictionRequest {
  Types.Image2D image = 1;
  double confidence = 2;
  Types.Region2D roi = 3;
}

/*
 * Responses
 *
 * Server --> Client
 */

message TrainingResponse {
  int32 label = 1;
  Types.Region2D face = 2;
}

message PredictionResponse {
  repeated int32 labels = 1;
  repeated double distances = 2;
  repeated Types.Region2D rois = 3;
}
